services/database/programcardinfo.ts

```diff
@@ -0,0 +1,168 @@
+    const findResults = await prisma.class.findMany({
+        where: {
+            programId: parseInt(id),
+            isArchived: false,
+        },
+        select: {
+            ageGroup: true,
+            spaceTotal: true,
+            id: true,
+            name: true,
+            spaceAvailable: true,
+            volunteerSpaceTotal: true,
+            volunteerSpaceAvailable: true,
+            startDate: true,
+            endDate: true,
+            weekday: true,
+            startTimeMinutes: true,
+            durationMinutes: true,
+            imageLink: true,
+
+            classTranslation: {
+                where: {
+                    language: locale.en, // TODO dont hardcode language
+                },
+                select: {
+                    name: true,
+                    description: true,
+                },
+            },
+            teacherRegs: {
+                select: {
+                    teacher: {
+                        select: {
+                            user: true,
+                        },
+                    },
+                },
+            },
+        },
+    });
+    return findResults.map((result) => {
+        return {
+            ageGroup: result.ageGroup,
+            spaceTotal: result.spaceTotal,
+            id: result.id,
+            image: result.imageLink,
+            spaceAvailable: result.spaceAvailable,
+            volunteerSpaceTotal: result.volunteerSpaceTotal,
+            volunteerSpaceAvailable: result.volunteerSpaceAvailable,
+            startDate: result.startDate,
+            endDate: result.endDate,
+            weekday: result.weekday,
+            startTimeMinutes: result.startTimeMinutes,
+            durationMinutes: result.durationMinutes,
+            name:
+                result.classTranslation.length > 0
+                    ? result.classTranslation[0].name
+                    : result.name,
+            description:
+                result.classTranslation.length > 0
+                    ? result.classTranslation[0].description
+                    : "",
+            teacherName:
+                result.teacherRegs.length > 0
+                    ? result.teacherRegs[0].teacher.user.firstName +
+                      " " +
+                      result.teacherRegs[0].teacher.user.lastName
+                    : "",
+        };
+    });
+}
+
+/**
+ * Get every program's card info
+ */
+async function getProgramCardInfos(): Promise<ProgramCardInfo[]> {
+    const findResult = await prisma.program.findMany({
+        select: {
+            id: true,
+            startDate: true,
+            endDate: true,
+            tag: true,
+            onlineFormat: true,
+            price: true,
+            imageLink: true,
+
+            programTranslation: {
+                where: {
+                    language: locale.en,
+                },
+            },
+        },
+    });
+    return findResult.map((p) => {
```

Comment:
Same comment as above

Reply:
The map doesn't really decrease amount of info, I included all the stuff in Program
The map is for flattening the name and description into the returned object, otherwise it'll have a nested programTranslation field
Program is 
```
  id SERIAL PRIMARY KEY NOT NULL,
  price INTEGER NOT NULL, -- price in cents, to make it integer
  online_format program_formats NOT NULL,
  tag TEXT NOT NULL, -- art, music, math etc (TODO we might need a tag table later)
  image_link TEXT,
  start_date TIMESTAMPTZ NOT NULL, -- earliest start date of all classes
  end_date TIMESTAMPTZ NOT NULL, -- latest end date of all classes
  is_archived BOOLEAN DEFAULT false NOT NULL,
  created_at TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
  updated_at TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP
```